src/type/definition.js

```diff
@@ -333,7 +334,7 @@ export class GraphQLScalarType {
   // Parses an externally provided value to use as an input.
   parseValue(value: mixed): mixed {
     const parser = this._scalarConfig.parseValue;
-    return parser && !isNullish(value) ? parser(value) : undefined;
+    return parser && !isNullish(value) ? parser(value) : value;
```

Comment:
If a value parses as nullish, shouldn't it return clearly as `undefined` to not be confused with a true `null` value?